packages/scanner/src/cli/scan/watchScan.ts

```diff
@@ -44,13 +47,54 @@ function isAncestorPath(ancestor: string, descendant: string): boolean {
   return !path.relative(ancestor, descendant).startsWith('..');
 }
 
+export type BatchedTelemetry = {
+  scanResults: ScanResults;
+  msElapsed: number;
+  msInserted: number;
+};
+
 export class Watcher {
   config?: TimestampedConfiguration;
   appmapWatcher?: chokidar.FSWatcher;
   appmapPoller?: chokidar.FSWatcher;
   configWatcher?: chokidar.FSWatcher;
+  batchedTelemetry: BatchedTelemetry[] = [];
+  sendPeriodMilliseconds = 5000;
+  interval: NodeJS.Timer;
+
+  constructor(private options: WatchScanOptions) {
+    this.interval = setInterval(() => {
+      this.sendScanResultsTelemetryInABatch();
+    }, this.sendPeriodMilliseconds);
+    // must unref because there's no condition under which the timer
+    // will stop; else the process doesn't exit
+    this.interval.unref();
+  }
 
-  constructor(private options: WatchScanOptions) {}
+  public sendScanResultsTelemetryInABatch() {
+    const dateNow = Date.now();
+    if (
+      this.batchedTelemetry &&
+      this.batchedTelemetry.length > 0 &&
+      dateNow - this.batchedTelemetry[0].msInserted > this.sendPeriodMilliseconds
+    ) {
+      const aggregatedScanResults: ScanResults = new ScanResults(
+        {
+          checks: [],
+        },
+        {},
+        [],
+        []
+      );
```

Comment:
This seems like a case for optional/default arguments in the constructor.

Reply:
Added default arguments.

Optional arguments had cascading effects that spread to `src/` and `tests/`